### PR TITLE
add datasourceUrl to export endpoint

### PR DESCRIPTION
**Pull request by MalteEbner**
## Description

The `ApiWorkflowClient.export_filenames_and_read_urls_by_tag_id` now also includes the datasource URL.

Other possible implementation options I tried but decided against:
1. As separate functions for the one with read_urls and datasource_urls -> lots of duplicate code.
2. By allowing to set the `FileNameFormat` -> makes it more complicated.

Why is the name of the method the same?
I know that it does not fit perfectly, but changing it would be breaking. Thus I would leave it as it is.

### CODE REVIEW

File: lightly/api/api_workflow_export.py [hunk 1]
```diff
@@ -237,19 +237,30 @@ def export_filenames_by_tag_name(
...
+                {
+                    "fileName": "sample2.jpg",
+                    "readURL": "s3://my_datasource/sample2.jpg?read_url_key=JSBFIEUHVSJ",
+                    "datasourceUrl": "s3://my_datasource/sample2.jpg",
+                },
+            ]
 
         """
-        # TODO (Philipp, 10.01.2023): Switch to the exportTagToBasicFilenamesAndReadUrls
-        # when the read-urls are fixed.
```
Comment (MalteEbner) [inline lightly/api/api_workflow_export.py:252]: That cannot be used for datasourceURLs

File: tests/api_workflow/test_api_workflow_export.py [hunk 2]
```diff
@@ -62,18 +64,34 @@ def test_export_label_studio_tasks_by_tag_name(self):
         self.assertTrue(all(isinstance(task, dict) for task in tasks))
 
     def test_export_tag_to_basic_filenames_and_read_urls(self):
```
Comment (MalteEbner) [inline tests/api_workflow/test_api_workflow_export.py:66]: I switched to not using the MockedApiWorkflowClient anymore.

File: lightly/api/api_workflow_export.py [hunk 3]
```diff
@@ -262,40 +273,50 @@ def export_filenames_and_read_urls_by_tag_id(
             tag_id=tag_id,
             file_name_format=FileNameFormat.REDIRECTED_READ_URL,
         )
+        datasource_urls_string = retry(
+            self._tags_api.export_tag_to_basic_filenames,
+            dataset_id=self.dataset_id,
+            tag_id=tag_id,
+            file_name_format=FileNameFormat.DATASOURCE_FULL,
+        )
```
Comment (guarin) [inline lightly/api/api_workflow_export.py:281]: What happens for this endpoint with old datasets that were created using the local workflow? My fear is that an empty string is returned which will result in `zip(filenames, read_urls, datasource_urls)` being and empty list.

File: lightly/api/api_workflow_export.py [hunk 4]
```diff
@@ -237,19 +237,30 @@ def export_filenames_by_tag_name(
     def export_filenames_and_read_urls_by_tag_id(
         self,
         tag_id: str,
-    ) -> List[Dict]:
-        """Export the samples filenames to map with their readURL.
+    ) -> List[Dict[str, str]]:
+        """Export the samples filenames to map with their readURL and datasource URL.
```
Comment (guarin) [inline lightly/api/api_workflow_export.py:241]:
```suggestion
        """Export filenames, read URLs, and datasource URLs from the given tag.
```

File: lightly/api/api_workflow_export.py [hunk 5]
```diff
@@ -237,19 +237,30 @@ def export_filenames_by_tag_name(
...
                 Id of the tag which should exported.
 
         Returns:
-            A list of mappings of the samples filenames and readURLs within a certain tag.
+            A list of dictionaries with the keys "filename", "readURL" and "datasourceURL".
+            An example:
+            [
+                {
+                    "fileName": "sample1.jpg",
+                    "readURL": "s3://my_datasource/sample1.jpg?read_url_key=EAIFUIENDLFN",
```
Comment (guarin) [inline lightly/api/api_workflow_export.py:253]:
```suggestion
                    "readUrl": "s3://my_datasource/sample1.jpg?read_url_key=EAIFUIENDLFN",
```

File: lightly/api/api_workflow_export.py [hunk 6]
```diff
@@ -262,40 +273,50 @@ def export_filenames_and_read_urls_by_tag_id(
...
+            )
         ]
 
     def export_filenames_and_read_urls_by_tag_name(
         self,
         tag_name: str,
-    ) -> List[Dict]:
-        """Export the samples filenames to map with their readURL.
+    ) -> List[Dict[str, str]]:
+        """Export the samples filenames to map with their readURL and datasource URL.
```
Comment (guarin) [inline lightly/api/api_workflow_export.py:304]:
```suggestion
        """Export filenames, read URLs, and datasource URLs from the given tag name.
```

File: lightly/api/api_workflow_export.py [hunk 7]
```diff
@@ -262,40 +273,50 @@ def export_filenames_and_read_urls_by_tag_id(
...
 
         Args:
             tag_name:
                 Name of the tag which should exported.
 
         Returns:
-            A list of mappings of the samples filenames and readURLs within a certain tag.
+            A list of dictionaries with the keys "filename", "readURL" and "datasourceURL".
```
Comment (guarin) [inline lightly/api/api_workflow_export.py:312]:
```suggestion
            A list of dictionaries with the keys "filename", "readUrl" and "datasourceUrl".

```

**Review by guarin — CHANGES_REQUESTED** [review]
Looks good :) There is one concern about datasets without a datasource.

File: lightly/api/api_workflow_export.py (hunk 3 above)
Comment (MalteEbner) [reply lightly/api/api_workflow_export.py:281]: It works, I tried it out for a dataset uploaded via CLI:
image

Comment (guarin) [reply lightly/api/api_workflow_export.py:281]: Awesome!

**Pull request merged**